app/Frontend/Conference/Pages/SubmissionDetail.php

```diff
@@ -17,7 +19,37 @@ class SubmissionDetail extends Page
     public function mount(int $submissionId)
     {
         $this->submission = Submission::find($submissionId);
-        abort_if(! $this->submission, 404);
+        
+        if (!$this->canAccess()) {
+            abort(404);
+        }
+    }
+
+    public function canAccess(): bool
+    {
+        return $this->submission->proceeding && 
+        (
+            (StageManager::editing()->isStageOpen() && auth()->user()?->can('editing', $this->submission)) || 
+            ($this->submission?->isPublished() && $this->submission->proceeding?->isPublished())
+        );
+    }
```

Comment:
Susah dipahami, breakdown jadi beberapa if agar mudah dipahami

Reply:
Kaya gini gimana bang?

    // Cek apakah submission memiliki proceeding
    if (!$this->submission->proceeding) {
        return false;
    }

    // Cek apakah stage editing terbuka dan pengguna memiliki izin untuk mengedit
    if (StageManager::editing()->isStageOpen() && auth()->user()?->can('editing', $this->submission)) {
        return true;
    }

    // Cek apakah submission dan proceeding dipublikasikan
    if ($this->submission->isPublished() && $this->submission->proceeding->isPublished()) {
        return true;
    }

    // Jika tidak memenuhi kondisi apapun di atas, kembalikan false
    return false;

Reply:
iya, kda ush pakai comment klo coding nya sudah jelas